public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -899,3 +905,24 @@ export const ignoreHiddenQueries = (ruleDefinition: RuleFormValues): RuleFormVal
 export function formValuesFromExistingRule(rule: RuleWithLocation<RulerRuleDTO>) {
   return ignoreHiddenQueries(rulerRuleToFormValues(rule));
 }
+
+export function getInstantFromDataQuery(model: AlertDataQuery, type: string): boolean | undefined {
+  // if the datasource is not prometheus or loki, instant is defined in the model or defaults to undefined
+  if (type !== DataSourceType.Prometheus && type !== DataSourceType.Loki) {
```

Comment:
I wonder if we should try to read the `instant` property for any other data source than Prometheus or Loki. I'm a bit worried it might lead to undefined behaviour as we don't know what other data sources might use that option for

What if we return an object like `{ instantCapable: boolean, isInstant: boolean}` and for each data source other than Prom and Loki `instantCapable` would return false meaning we should not try to make any optimizations

wdyt?

Reply:
For the current usage of this method this shouldn't be a problem, but I agree that being a public util method, we should be more generic and take in account other data sources 👍

Reply:
> I wonder if we should try to read the instant property for any other data source than Prometheus or Loki. I'm a bit worried it might lead to undefined behaviour as we don't know what other data sources might use that option for

The method is returning the instant property from the data query. I would say that if we do so (returning false in the current undefined), we are changing the current behaviour where these data sources don't contain this field in the data query.

Reply:
Okay, looks good. I thought the function should only work for Promehteus or Loki, but it's more generic